### ISSUE TITLE
Unable to pull from sregistry

### ISSUE THREAD

**Issue by brisk022** [sylabs/singularity#698]
**Version of Singularity**
3.9.x

**Describe the bug**
When I try to pull an image from sregistry, I get "unexpected HTTP status 302" error.

**To Reproduce**
```
singularityce Ф❯ singularity --version
singularity-ce version 3.9.7
singularityce Ф❯ singularity pull --disable-cache library://rbrisk/apps/test:2.0
INFO:    Downloading library image
FATAL:   While pulling library image: error fetching image: unable to download image: error downloading image: unexpected HTTP status 302: <nil>
```

**Expected behavior**
Everything works fine with SingularityCE < 3.9.0. For example, here is the output from 3.7.1.
```
singularity Ф❯ singularity --version
singularity version 3.7.1
singularity Ф❯ singularity pull --disable-cache library://rbrisk/apps/test:2.0
INFO:    Downloading library image
58.0MiB / 58.0MiB [==============================================================================] 100 % 57.5 MiB/s 0s
WARNING: integrity: signature not found for object group 1
WARNING: Skipping container verification
```

**OS / Linux Distribution**
Ubuntu, Manjaro, Arch

**Installation Method**
conda (Manjaro, Arch), deb (Ubuntu)

**Additional context**
The problem appeared after the switch to ConcurrentDownloadImage in 3.9.0. I think it might be fixed if the following enhancement is implemented: https://github.com/sylabs/scs-library-client/issues/130

**Comment by vsoch** [comment]
+1 this does look like it could be the culprit! https://github.com/sylabs/scs-library-client/issues/130

**Comment by brisk022** [comment]
A potential fix might be to change the custom [CheckRedirect function](https://github.com/sylabs/scs-library-client/blob/36cf52782fb0701fb9c3277bac4d37b37bb5a881/client/pull.go#L253l) to

```
CheckRedirect: func(req *http.Request, via []*http.Request) error {
	if res.StatusCode == http.StatusSeeOther {
		return http.ErrUseLastResponse
	}
	if len(via) >= 10 {
		return errors.New("stopped after 10 redirects")
	}
	return nil
},
```
Instead of the length check, it would have been better to call `defaultCheckRedirect` from `http/client.go` but I suspect it is private. So, it is probably not possible. (Sorry, I do not have any golang experience.)

I will try to wrap it into a pull request if I can figure out how to compile and test it.

**Comment by brisk022** [comment]
I have submitted a pull request. I tested it manually pulling from sregistry, docker, and shub. All seemed to work properly.

**Comment by dtrudg** [comment]
Thanks @brisk022 - I've brought this to the attention of the main maintainers of the library client. Note that `docker` and `shub` sources don't actually use that code, it's just for `library` URIs. 

It may take some time to look at it, as changes here have to be verified against all deployed versions of Sylabs Singularity Enterprise product that provides an on-premise library, as well as the free cloud services.

**Comment by tri-adam** [comment]
Thanks for the fix @brisk022, it has been released with [github.com/sylabs/scs-library-client@v1.3.1](https://github.com/sylabs/scs-library-client/releases/tag/v1.3.1) and will make its way into Singularity shortly.

**Comment by vsoch** [comment]
woohoo! Thank you @tri-adam ! 

I tried searching for a "thank you" emoji but the first one that came back is "turkey" and now I simply must use it!! :laughing: 

:turkey:

**Comment by tri-adam** [comment]
My pleasure. Gobble gobble.

**Comment by brisk022** [comment]
Thanks a lot for addressing it so quickly!

**Comment by tri-adam** [comment]
Fixed in #701 (`release-3.9`), #702 (`master`).
